controllers/managedmcg_controller_test.go

```diff
@@ -36,88 +38,139 @@ import (
-
-	return ODFCSVFake
+	return managedMCGFake
 }
 
-func newManagedMCGFake() mcgv1alpha1.ManagedMCG {
-	managedMCGFake := mcgv1alpha1.ManagedMCG{
+func newAddonSecretFake() corev1.Secret {
+	secretFake := corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "managedmcgfake",
+			Name:      "AddOnSecretfake",
 			Namespace: "openshift-storage",
-			UID:       "fake-uid",
 		},
-		Spec: mcgv1alpha1.ManagedMCGSpec{
-			ReconcileStrategy: "ignore",
+	}
+	secretFake.Data = make(map[string][]byte, 1)
+	secretFake.Data["addonparam"] = []byte("foo")
+	return secretFake
+}
+
+func newPagerDutySecretFake() corev1.Secret {
+	secretFake := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "PagerDutySecretfake",
+			Namespace: "openshift-storage",
 		},
 	}
+	secretFake.Data = make(map[string][]byte, 1)
+	secretFake.Data["PAGERDUTY_KEY"] = []byte("foo")
+	return secretFake
+}
 
-	return managedMCGFake
+func newDeadMansSecretFake() corev1.Secret {
+	secretFake := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "DeadMansSecretfake",
+			Namespace: "openshift-storage",
+		},
+	}
+	secretFake.Data = make(map[string][]byte, 1)
+	secretFake.Data["SNITCH_URL"] = []byte("foo")
+	return secretFake
 }
 
+func newSmtpSecretFake() corev1.Secret {
+	secretFake := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "SmtpSecretfake",
+			Namespace: "openshift-storage",
+		},
+	}
+	secretFake.Data = make(map[string][]byte, 1)
+	secretFake.Data["host"] = []byte("host")
+	secretFake.Data["port"] = []byte("8080")
+	secretFake.Data["username"] = []byte("username")
+	secretFake.Data["password"] = []byte("password")
+	return secretFake
+}
+func newOcsCsvFake() opv1a1.ClusterServiceVersion {
+	ocscsv := opv1a1.ClusterServiceVersion{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "ocs-operator",
+			Namespace: "openshift-storage",
+		},
+	}
+	return ocscsv
+}
+func cleanup() {
+	os.Remove(CUST_FILE)
+}
 func TestManagedMCGReconcilerReconcile(t *testing.T) {
 	r := &ManagedMCGReconciler{}
 	r.Log = ctrl.Log.WithName("controllers").WithName("ManagedMCGFake")
+
 	r.Scheme = newSchemeFake()
-	r.initializeReconciler(reconcile.Request{
-		NamespacedName: types.NamespacedName{
-			Name:      "managedmcgfake",
-			Namespace: "openshift-storage",
-		},
-	})
-	ODFCSVFake := newODFCSVFake()
+	ocscsvFake := newOcsCsvFake()
 	managedMCGFake := newManagedMCGFake()
-	fakeClient := fake.NewClientBuilder().WithScheme(r.Scheme).WithObjects(&ODFCSVFake, &managedMCGFake).Build()
+	addonsecretFake := newAddonSecretFake()
+	smtpsecretfake := newSmtpSecretFake()
+	pagerdutysecretFake := newPagerDutySecretFake()
+	deadmansercretFake := newDeadMansSecretFake()
+
+	r.AddonParamSecretName = "AddOnSecretfake"
+	r.DeadMansSnitchSecretName = "DeadMansSecretfake"
+	r.PagerdutySecretName = "PagerDutySecretfake"
+	r.SMTPSecretName = "SmtpSecretfake"
+	r.CustomerNotificationHTMLPath = CUST_FILE
+
+	data := []byte{}
+	err := os.WriteFile(CUST_FILE, data, 0644)
```

Comment:
Minor suggestion: `644` translates to `110100100`,i.e.,`rw-r--r--` for the owner, group and all users. If we are not writing to this file in the future, `444` might be the better choice.

Reply:
Done.